packages/utils/src/baggage.ts

```diff
@@ -0,0 +1,54 @@
+export type AllowedBaggageKeys = 'environment' | 'release'; // TODO: Add remaining allowed baggage keys | 'transaction' | 'userid' | 'usersegment';
+export type Baggage = Partial<Record<AllowedBaggageKeys, string> & Record<string, string>>;
```

Comment:
Probably a dumb question but what's the advantage of making this a record instead of creating e..g a `Baggage` interface with the baggage keys as properties?

Reply:
A record type is just sugar over creating a interface: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type - this just makes is so that to add additional keys we just need to add to the union type, not to the interface.

Reply:
Jumping on this thread, I would vote for just making this an interface if it's not too much of an effort to change. Currently it is quite hard to read.

Reply:
```ts
type BaggageObj = Partial<Record<AllowedBaggageKeys, string> & Record<string, string>>;
```

desugars down to

```ts
type BaggageObj = {
    // comes from Record<string, string>
    [x: string]: string | undefined;
    // comes from Record<AllowedBaggageKeys, string>
    environment?: string | undefined;
    release?: string | undefined;
}
```

IMO the union type makes it clear how the two object types (created by the `Record` type) will come together to form the final object structure. In addition, users have to use the functions to interact with baggage - it should be sufficiently abstracted from them.